lib/__tests__/fixtures/plugin-meta-fixer.mjs

```diff
@@ -0,0 +1,59 @@
+import stylelint from '../../index.mjs';
+
+const {
+	createPlugin,
+	utils: { report, ruleMessages, validateOptions },
+} = stylelint;
+
+const ruleName = 'plugin/selector-no-foo';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (selector) => `Unexpected "foo" in selector "${selector}"`,
+});
+
+const rule = (primary, _, context) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: [true],
+		});
+
+		if (!validOptions) return;
+
+		root.walkRules((ruleNode) => {
+			const { selector } = ruleNode;
+			const i = selector.indexOf('foo');
+
+			if (i === -1) return;
+
+			const range = ruleNode.rangeBy({
+				index: i,
+				endIndex: i + 'foo'.length,
+			});
+			let fix;
+
+			if (context.fix) {
+				fix = () => {
+					ruleNode.selector = selector.replace('foo', 'bar');
+
+					return range;
+				};
+			}
+
+			report({
+				result,
+				ruleName,
+				message: messages.rejected(selector),
+				node: ruleNode,
+				...range,
+				fix,
+			});
```

Comment:
```suggestion
			report({
				result,
				ruleName,
				message: messages.rejected(selector),
				node: ruleNode,
				...range,
				fix: () => {
					ruleNode.selector = selector.replace('foo', 'bar');

					return range;
				},
			});
```

You can internalize the logic on when to apply `fix`.

If you do, you can remove the flag/option added in this PR.
This was part of the motivation behind : https://github.com/stylelint/stylelint/pull/7733

Instead of requiring users to set a flag you can dynamically pick the expected behavior.

----

Is it fine if I demonstrate how this could work with in a pull request?

Reply:
So you wanna pass `fix`, `unfixable` _and_ `context` (or `context.fix`) to `report`?
If you meant doing
```
				fix: context.fix ? (() => {
					ruleNode.selector = selector.replace('foo', 'bar');
					return range;
				}) : undefined,
```
instead, I don't think that would be cleaner.
Also, you have rules with multiple fixers so inlining wouldn't be recommended in that case.

> Is it fine if I demonstrate how this could work in a pull request?

I would prefer if you opened another PR.

Reply:
I've opened another PR to discuss what I had in mind : https://github.com/stylelint/stylelint/pull/7741